### PR TITLE
nested bdb_thread_event calls, only last frees

### PR DESCRIPTION
**Pull request by dorinhogea**
Trace:
"delete_thread_lock_info: thread stop before init!" 
(shows on LOGMSG_WARN level).
There are cases when we acquire first the bdb lock in read mode, and later on we switch to write mode.  Each call prepares the call tracing info by calling bdb_thread_event, and frees the tracing with a secondary call.
When calls are nested, the second free call believes the thread was never initialized.
The patch is tracking and freeing the locking tracing structure on the last release call.

### CODE REVIEW

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Error &#x26A0;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>6/487 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>systable_locking<br>sc_truncate<br>phys_rep<br>phys_rep_tiered<br>queuedb_rollover<br>sp_snapshot_generated</p>

File: bdb/bdblock.c [hunk 1]
```diff
@@ -804,6 +806,16 @@ static void delete_thread_lock_info(bdb_state_type *bdb_state)
         abort_lk(lk);
     }
 
+    if (lk->callers > 1) {
+        /* if we have called bdb_thread_event nested, only
+         * the last caller frees it
+         * NOTE: obviously, only the first stack is saved ! 
+         */
```
Comment (markhannum) [inline bdb/bdblock.c:813]: Should this decrement lk->callers?

Comment (dorinhogea) [reply bdb/bdblock.c:815]: duh, fixed that

**Review by markhannum — APPROVED** [review]

**Comment by dorinhogea** [comment]
Thank you Mark!

**Pull request merged**